syntax = 'proto3';

package o2.bookkeeping;

import 'common.proto';

service QcFlagService {
  rpc CreateForDataPass(DataPassQcFlagCreationRequest) returns (QcFlagCreationResponse);
  rpc CreateForSimulationPass(SimulationPassQcFlagCreationRequest) returns (QcFlagCreationResponse);
}

message QcFlagCreationResponse {
  repeated int32 flagIds = 1;
}

message DataPassQcFlagCreationRequest {
  int32 runNumber = 1;
  // usually apassN, cpassN, where N is a natural number
  string passName = 2;
  string detectorName = 3;
  repeated QcFlag flags = 4;
}

message SimulationPassQcFlagCreationRequest {
  int32 runNumber = 1;
  // Simulation Pass (production) Tag
  string productionName = 2;
  string detectorName = 3;
  repeated QcFlag flags = 4;
}

message QcFlag {
  uint32 flagTypeId = 1;
  // Unix timestamp (ms) since which QC flag applies to data taken during given run
  optional uint64 from = 2;
  // Unix timestamp (ms) until which QC flag applies to data taken during given run
  optional uint64 to = 3;
  // string indicating why a given flag was given, providing some details about the nature of the problem
  optional string comment = 4;
  // string indicating what produced this flag.
  // it could be a check name (e.g. TPC/Clusters, ITS/Efficiency) or a person name
  string origin = 5;
}
